app/javascript/mastodon/components/column_header.tsx

```diff
@@ -0,0 +1,309 @@
+import { withRouter } from 'react-router-dom';
+
+import { ReactComponent as AddIcon } from '@material-symbols/svg-600/outlined/add.svg';
+import { ReactComponent as ArrowBackIcon } from '@material-symbols/svg-600/outlined/arrow_back.svg';
+import { ReactComponent as ChevronLeftIcon } from '@material-symbols/svg-600/outlined/chevron_left.svg';
+import { ReactComponent as ChevronRightIcon } from '@material-symbols/svg-600/outlined/chevron_right.svg';
+import { ReactComponent as CloseIcon } from '@material-symbols/svg-600/outlined/close.svg';
+import { ReactComponent as TuneIcon } from '@material-symbols/svg-600/outlined/tune.svg';
+
+import type { IconProp } from 'mastodon/components/icon';
+import { Icon } from 'mastodon/components/icon';
+import {
+  ButtonInTabsBar,
+  useColumnsContext,
+} from 'mastodon/features/ui/util/columns_context';
+import type { ContextWithIdentity } from 'mastodon/utils/identity';
+
+import { useAppHistory } from './router';
+
+const messages = defineMessages({
+  show: { id: 'column_header.show_settings', defaultMessage: 'Show settings' },
+  hide: { id: 'column_header.hide_settings', defaultMessage: 'Hide settings' },
+  moveLeft: {
+    id: 'column_header.moveLeft_settings',
+    defaultMessage: 'Move column to the left',
+  },
+  moveRight: {
+    id: 'column_header.moveRight_settings',
+    defaultMessage: 'Move column to the right',
+  },
+});
+
+interface BackButtonProps {
+  pinned?: boolean;
+  show?: boolean;
+}
+
+const BackButton = ({ pinned, show }: BackButtonProps) => {
+  const history = useAppHistory();
+  const { multiColumn } = useColumnsContext();
+
+  const handleBackClick = useCallback(() => {
+    if (history.location.state?.fromMastodon) {
+      history.goBack();
+    } else {
+      history.push('/');
+    }
+  }, [history]);
+
+  const showButton =
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    history &&
+    !pinned &&
+    ((multiColumn && history.location.state?.fromMastodon) || show);
+
+  if (!showButton) return null;
+
+  return (
+    <button onClick={handleBackClick} className='column-header__back-button'>
+      <Icon
+        id='chevron-left'
+        icon={ArrowBackIcon}
+        className='column-back-button__icon'
+      />
+      <FormattedMessage id='column_back_button.label' defaultMessage='Back' />
+    </button>
+  );
+};
+
+interface ColumnHeaderProps extends RouteComponentProps {
+  active?: boolean;
+  appendContent?: React.ReactNode;
+  children?: React.ReactNode;
+  collapseIssues?: boolean;
+  extraButton?: React.ReactNode;
+  icon?: string;
+  iconComponent?: IconProp;
+  intl: IntlShape;
+  multiColumn?: boolean;
+  onClick?: () => void;
+  onMove?: (direction: number) => void;
+  onPin?: () => void;
+  pinned?: boolean;
+  placeholder?: boolean;
+  showBackButton?: boolean;
+  title?: React.ReactNode;
+}
+
+class ColumnHeaderInternal extends PureComponent<ColumnHeaderProps> {
+  declare context: ContextWithIdentity;
```

Comment:
This bit of TypeScript-specific syntax is giving Jest grief:

```plaintext
         97 | class ColumnHeaderInternal extends PureComponent<ColumnHeaderProps> {
      >  98 |   declare context: ContextWithIdentity;
            |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It looks to me like Jest isn't fully configured to be able to transpile TypeScript syntax. Would you like me to tackle that in this PR or a separate one?

Reply:
This is weird, Jest should be using the same Babel config as Webpack.

Do you have an idea where this comes from?

Reply:
I'm not super familiar with how the transforms pipeline are handled here. But adding to `overrides` in `babel.config.js`'s `case 'test'` is a way to do it.

Reply:
Converting it to a function component bypassed this need.